src/core/namespace.c

```diff
@@ -1892,6 +1898,49 @@ static int apply_one_mount(
         }
 
         log_debug("Successfully mounted %s to %s", what, mount_entry_path(m));
+
+        /* Take care of id-mapped mounts */
+        if (m->idmapped && uid_is_valid(m->uid) && gid_is_valid(m->gid)) {
+                _cleanup_close_ int userns_fd = -EBADF;
+                _cleanup_free_ char *uid_map = NULL, *gid_map = NULL;
+
+                log_debug("Setting an id-mapped mount on %s", mount_entry_path(m));
+
+                /* Do mapping from nobody (in setup_exec_directory()) -> this uid */
+                r = strextendf(&uid_map, UID_FMT " " UID_FMT " " UID_FMT "\n", UID_NOBODY, (uid_t)m->uid, 1u);
+                if (r < 0)
+                        return log_oom();
```

Comment:
you don't really care about `r` here, which is fine, hence i would shorten these lines to:

```c
if (strextendf(…) < 0)
        return log_oom()
```